crates/consensus/src/engine/mod.rs

```diff
@@ -189,18 +189,22 @@ impl<Client: HeaderProvider + BlockProvider + StateProvider> ConsensusEngine
              return Ok(PayloadStatus::from_status(PayloadStatusEnum::Syncing))
         };
 
-        let parent_td = self.client.header_td(&block.parent_hash)?;
-        if parent_td.unwrap_or_default() <= self.config.merge_terminal_total_difficulty.into() {
-            return Ok(PayloadStatus::from_status(PayloadStatusEnum::Invalid {
-                validation_error: EngineApiError::PayloadPreMerge.to_string(),
-            }))
+        if let Some(parent_td) = self.client.header_td(&block.parent_hash)? {
+            if parent_td <= self.config.merge_terminal_total_difficulty.into() {
+                return Ok(PayloadStatus::from_status(PayloadStatusEnum::Invalid {
+                    validation_error: EngineApiError::PayloadPreMerge.to_string(),
```

Comment:
perhaps we can make `validation_error: Box< dyn std::error::Error + Send + Sync` instead?

Reply:
Agree